source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -793,7 +794,8 @@ namespace
             {
                 // Write input if it's connected or different from default value.
                 if (writeDefaultValues || 
-                    input.isConnected() || !RtValue::compare(input.getType(), input.getValue(), attrDef.getValue()))
+                    input.isConnected() || !RtValue::compare(input.getType(), input.getValue(), attrDef.getValue()) ||
+                    (input.getMetadata(UI_VISIBLE) && input.getMetadata(UI_VISIBLE)->getValueString() == "false"))
```

Comment:
I'm not sure we should ignore writing the input to file just because it's hidden in the UI. I think there is a risk of loosing data. If the input had a value set that was different from it's default value, and a user decide to hide the input in the UI, the new value will be lost since it's not written out.

As long as we write out the `uivisible` metadata (which we should by default), all information is preserved.